pkg/lib/synthesize/resource_getters.go

```diff
@@ -0,0 +1,240 @@
+}
+
+func (s *synthesize) getClusterRoleBindings() ([]rbacv1.ClusterRoleBinding, error) {
+	if crbIF, ok := s.unstructuredObjectMap[ClusterRoleBindingKind]; ok {
+		var crbs []rbacv1.ClusterRoleBinding
+		for _, bufObj := range crbIF {
+			crb := rbacv1.ClusterRoleBinding{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(bufObj.UnstructuredContent(), &crb); err != nil {
+				return nil, fmt.Errorf("error decoding %v as %s kind, %v", bufObj.GroupVersionKind(), bufObj.GetKind(), err)
+			}
+			crbs = append(crbs, crb)
+		}
+		return crbs, nil
+	}
+	return nil, ObjectTypeNotFoundInBundleError
+}
+
+func (s *synthesize) getServices() ([]corev1.Service, error) {
+	if serviceIF, ok := s.unstructuredObjectMap[ServiceKind]; ok {
+		var services []corev1.Service
+		for _, bufObj := range serviceIF {
+			service := corev1.Service{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(bufObj.UnstructuredContent(), &service); err != nil {
+				return nil, fmt.Errorf("error decoding %v as %s kind, %v", bufObj.GroupVersionKind(), bufObj.GetKind(), err)
+			}
+			services = append(services, service)
+		}
+		return services, nil
+	}
+	return nil, ObjectTypeNotFoundInBundleError
+}
+
+func (s *synthesize) getSAs() ([]corev1.ServiceAccount, error) {
+	if saIF, ok := s.unstructuredObjectMap[ServiceAccountKind]; ok {
+		var sas []corev1.ServiceAccount
+		for _, bufObj := range saIF {
+			sa := corev1.ServiceAccount{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(bufObj.UnstructuredContent(), &sa); err != nil {
+				return nil, fmt.Errorf("error decoding %v as %s kind, %v", bufObj.GroupVersionKind(), bufObj.GetKind(), err)
+			}
+			sas = append(sas, sa)
+		}
+		return sas, nil
+	}
+	return nil, ObjectTypeNotFoundInBundleError
+}
+
+func (s *synthesize) getRoles() ([]rbacv1.Role, error) {
+	if roleIF, ok := s.unstructuredObjectMap[RoleKind]; ok {
+		var roles []rbacv1.Role
+		for _, bufObj := range roleIF {
+			role := rbacv1.Role{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(bufObj.UnstructuredContent(), &role); err != nil {
+				return nil, fmt.Errorf("error decoding %v as %s kind, %v", bufObj.GroupVersionKind(), bufObj.GetKind(), err)
+			}
+			roles = append(roles, role)
+		}
+		return roles, nil
+	}
+	return nil, ObjectTypeNotFoundInBundleError
+}
+
+func (s *synthesize) getRoleBindings() ([]rbacv1.RoleBinding, error) {
+	if rbIF, ok := s.unstructuredObjectMap[RoleBindingKind]; ok {
+		var rbs []rbacv1.RoleBinding
+		for _, bufObj := range rbIF {
+			rb := rbacv1.RoleBinding{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(bufObj.UnstructuredContent(), &rb); err != nil {
+				return nil, fmt.Errorf("error decoding %v as %s kind, %v", bufObj.GroupVersionKind(), bufObj.GetKind(), err)
+			}
+			rbs = append(rbs, rb)
+		}
+		return rbs, nil
+	}
+	return nil, ObjectTypeNotFoundInBundleError
+}
+
+func (s *synthesize) getDeployments() ([]appsv1.Deployment, error) {
```

Comment:
I'm desperate for an implementation of this file that doesn't need to duplicate itself so much. Maybe later?